Core/src/IfcGeometryConverter/GeomUtils.h

```diff
@@ -62,93 +62,286 @@ namespace OpenInfraPlatform
+				*
+				* \param[in] points_vec				A vector of points. 
+				*
+				* \param[out] target_vec			A curve. 
+				*/
 				static void appendPointsToCurve(
-					const std::vector<carve::geom::vector<3> >& points_vec,
-					std::vector<carve::geom::vector<3> >& target_vec);
+					const std::vector<carve::geom::vector<3>>& points_vec,
+					std::vector<carve::geom::vector<3>>& target_vec);
+
+				/**********************************************************************************************/
+				/*! \brief Appedns vector of points (in 2D) to the curve.
+				*
+				* \param[in] points_vec				A vector of points.
+				*
+				* \param[out] target_vec			A curve.
+				*/
 				static void appendPointsToCurve(
-					const std::vector<carve::geom::vector<2> >& points_vec,
-					std::vector<carve::geom::vector<3> >& target_vec);
-
+					const std::vector<carve::geom::vector<2>>& points_vec,
+					std::vector<carve::geom::vector<3>>& target_vec);
+
+				/**********************************************************************************************/
+				/*! \brief Computes an inverse of the matrix.
+				*
+				* \param[in] matrix_a				An initial invertible matrix.
+				*
+				* \return							Inverted matrix
+				*/
 				static carve::math::Matrix computeInverse(
 					const carve::math::Matrix& matrix_a);
 
+				/**********************************************************************************************/
+				/*! \brief Calculates a closest point on the line to the point. 
+				*
+				* \param[in] point					Point outside the line. 
+				* \param[in] line_origin			Line origin point. 
+				* \param[in] line_direction			Direction of the line. 	
+				*
+				* \param[out] closest				Closest point on the line to the point.
+				*/
 				static void closestPointOnLine(const carve::geom::vector<3>& point,
 					const carve::geom::vector<3>& line_origin,
 					const carve::geom::vector<3>& line_direction,
 					carve::geom::vector<3>& closest);
 
-
+				/**********************************************************************************************/
+				/*! \brief Calculates a closest point on the line to the point.
+				*
+				* \param[in] point					Point outside the line.
+				* \param[in] line_origin			Line origin point.
+				* \param[in] line_direction			Direction of the line.
+				*
+				* \param[out] closest				Closest point on the line to the point.
+				*/
 				static void closestPointOnLine(const buw::Vector3f& point,
 					const buw::Vector3f& line_origin,
 					const buw::Vector3f& line_direction,
 					buw::Vector3f& closest);
 
+				/**********************************************************************************************/
+				/*! \brief Verify if point is on the segment of the line. 
+				*
+				* \param[in] lambda					Lambda of the line. 
+				* \param[in] point					Point which should be verified. 
+				* \param[in] line_origin			Line origin point.
+				* \param[in] line_direction			Direction of the line.
+				*

```

Comment:
I'd include an example to make this comment more understandable (I believe there should be an `\example` key for doxycomments): `lambda = 0` means the point equals `line_origin`, while `lambda=1` equals `line_origin + line_direction`.

Reply:
![image](https://user-images.githubusercontent.com/62873079/121817297-ce107f80-cc80-11eb-8274-c853ca13ea6a.png)

I guess \example is used in Doxygen a little bit differently. I will add this example with \note

Reply:
Resolved: 36ad1fc

Reply:
note is fine as well!